Add hooks for external timesheet plugin

piotrprusinski opened the pull request:
```
* button accept raport
```

Commit:
Add hooks for external timesheet plugin
	* button accept raport

mgierulski commented on app/views/hooks_view/_view_plugin_timesheet_views_timesheets_report_before_time_entries.html.erb:
```diff
@@ -0,0 +1,16 @@
+<%= submit_tag 'Akceptuj zaznaczone', :id=> 'timeEntriesAccept' -%>
```
nie, żeby to było dla nas konieczne, ale skoro plugin jest w ogólnodostępnym repo, to może by napisy wypisywać z jakichś internacjonalizowanych propertisów?

mgierulski commented on app/views/hooks_view/_view_plugin_timesheet_views_timesheets_report_before_time_entries.html.erb:
```diff
@@ -0,0 +1,16 @@
+	$(document).ready(function() {
+		$("#timeEntriesAccept").click(function() {
+			$("#time_entries").append($(document.createElement("input")).attr({
+				'id' : 'time_entry_accepted_raport',
```
Po angielsku nie jest raport, tylko report (dotyczy całego kodu...)

mgierulski commented on db/migrate/152_update_time_entries.rb:
```diff
@@ -0,0 +1,15 @@
+class UpdateTimeEntries < ActiveRecord::Migration
+  def self.up
+    change_table :time_entries do |t|
```
docelowo dołożyłbym jeszcze datę akceptacji

Commit:
patch set #1

mgierulski commented on lib/controllers_patch/acceptreport_controller.rb:
```diff
@@ -0,0 +1,62 @@
+          time_entry.safe_attributes = attributes
+
+          if(paramsHasAcceptParameter(params))
+            logger.info time_entry.accepted_report
```
Co tu się dokładnie odkłada w logu? Czy to jest debug, czy wartościowa informacja? Np, że ten user akceptuje ten time_entry wtedy i wtedy?

piotrprusinski replied:
to był debug , usunę to

mgierulski replied:
W takim razie może dobrze wrzucić do info to, o czym pisałem wcześniej - że ten user akceptuje to time entry w tym momencie.

piotrprusinski replied:
myślę że "to time_entry" by było trochę obciążające przy całym dużym raporcie. zaloguję to raz przy jednym kliknięciu akceptacji

mgierulski commented on lib/controllers_patch/acceptreport_controller.rb:
```diff
@@ -0,0 +1,62 @@
+            if(!time_entry.accepted_report)
+              time_entry.update_column("accepted_report", 1)
+              time_entry.update_column("accepted_report_user", User.current.id)
+              time_entry.update_column("accepted_report_date", Time.now.utc)
```
Ja bym wszystkim ustawił tą samą datę akceptacji, czyli wyliczył Time.now.utc wcześniej.
Na pewno chcemy utc, a nie datę z naszej lokalizacji?

mgierulski commented on lib/controllers_patch/acceptreport_controller.rb:
```diff
@@ -0,0 +1,62 @@
+          time_entry.reload
+          time_entry.safe_attributes = attributes
+
+          if(paramsHasAcceptParameter(params))
```
Wiem, że to bez znaczenia, ale warto wołać tą funkcję tyle razy, a nie raz przed pętlą?

piotrprusinski replied:
faktycznie pod kątem optymalizacji ta funkcja jest beznadziejna , poprawię to

Commit:
patch set #3
	* add back_url after accept report
	* optymalization code
	* add log that user accept report

Commit:
remove unused hooks

Commit:
Add block change tilme_entry if is accepted by report

Commit:
polish translate label

Commit:
Add filtr whose select accepted or not accepted time entries
	*add column for seen whose time_entry is accepted and by who and wheAdd filter whose select accepted or not accepted time entries	*add column for seen whose time_entry is accepted and by who and whenn

Commit:
Add permission to button accept_report

Commit:
Remove strong tag from time_entry details

mgierulski commented on app/views/hooks_view/_view_plugin_timesheet_view_timesheet_form.html.erb:
```diff
@@ -0,0 +1,7 @@
+<div class="radiobutton-label">
+	<label for="accepted_report"><%= l(:timesheet_filtr_by_report)%></label>
+	<br><%= radio_button_tag 'timesheet[accepted_report]', 'all', true%> <%= l(:timesheet_filtr_display_all)%>
```
Jak to po angielsku, to filter, a nie filtr ;)

mgierulski commented on config/locales/en.yml:
```diff
@@ -1,3 +1,12 @@
-  # my_label: "My label"
+  accept_not_accepted: Accept not yet accepted
+  report_is_accpeted: time entry is accepted by report, you cannot change it
+  timesheet_filtr_by_accepted_report: timesheet_filtr_by_accepted_report
```
moze po prostu 'accepted'

mgierulski commented on config/locales/pl.yml:
```diff
@@ -0,0 +1,12 @@
+pl:
+  accept_not_accepted: Akceptuj nie zaakceptowane
+  report_is_accpeted: Czas został już zaakceptowny w raporcie
+  timesheet_filtr_by_accepted_report: timesheet_filtr_by_accepted_report
```
'Zaakceptowane'

mgierulski commented on config/locales/en.yml:
```diff
@@ -1,3 +1,12 @@
+  timesheet_filtr_by_report: Report
+  redmine_timesheet_accept_button: Accept Report timesheet
+  is_accepted: Accepted
+  accepted_report_user: User accepted report
```
Accepted by

mgierulski commented on config/locales/pl.yml:
```diff
@@ -0,0 +1,12 @@
+  timesheet_filtr_by_report: Raport
+  redmine_timesheet_accept_button: Akceptuj raport
+  is_accepted: Zaakceptowany
+  accepted_report_user: Użytkownik zaakceptował
```
Zaakceptowane przez

mgierulski commented on lib/controllers_patch/acceptreport_controller.rb:
```diff
@@ -0,0 +1,71 @@
+          @time_entries.each do |time_entry|
+            time_entry.reload
+
+            if(!time_entry.accepted_report)
```
Dodałbym tu wywołanie metody sprawdzającej, czy user może zaakceptować to konkretne time entry. A w tej metodzie - na razie - sprawdzenie, czy user ma odpowiednie uprawnienie. Później tam się wrzuci jakąś logikę, czy ten konkretny user może akceptować to konkretne entry.
Chyba, że to nie po redminowemu?

mgierulski commented on lib/controllers_patch/acceptreport_controller.rb:
```diff
@@ -0,0 +1,71 @@
+
+          end
+
+          logger.info "User '#{User.current.login}' accept report at #{acceptedReportDate}"
```
Jeśli to ma służyć jakiemuś śledzeniu informacji, to trzeba podać, jakie entry są akceptowane (a więc może w pętli do każdego entry to zalogować?)

piotrprusinski replied:
moim zdanie to by było nie potrzebne obciążenie. myślę że po dacie dojdziemy które time_entry zostały zaakceptowane przez tego usera

mgierulski commented on lib/controllers_patch/acceptreport_controller.rb:
```diff
@@ -0,0 +1,71 @@
+          end
+
+          logger.info "User '#{User.current.login}' accept report at #{acceptedReportDate}"
+        else
```
czyli przy akceptacji nie zmieniają się inne dane?

piotrprusinski replied:
jeżeli jest parametr accepted_report to inne dane nie są zmieniane w innym wypadku są edytowane ...

mgierulski commented:
:+1: zostawiłem parę komentarzy, ale to nie błędy, tylko zmiany, które wg mnie nie blokują merge tego co jest

piotrprusinski committed:
Merge pull request #2 from AgoraTech/permision-branch

Add permission to button accept_report

Commit:
patch set #1
